packages/fxa-admin-panel/src/components/PageRateLimiting/index.tsx

```diff
@@ -0,0 +1,270 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useLazyQuery, useMutation, ApolloError } from '@apollo/client';
+import iconSearch from '../../images/icon-search.svg';
+import ErrorAlert from '../ErrorAlert';
+import { BlockStatus } from './BlockStatus';
+import { GET_RATE_LIMITS, UNBLOCK, UNBAN } from './index.gql';
+import { BlockStatus as BlockStatusData } from 'fxa-admin-server/src/graphql';
+
+interface RateLimitSearchState {
+  ip: string;
+  email: string;
+  uid: string;
+}
+
+type RateLimitQuery = Partial<RateLimitSearchState>;
+
+export const PageRateLimiting = () => {
+  const { register, handleSubmit, reset, getValues } =
+    useForm<RateLimitSearchState>({
+      defaultValues: {
+        ip: '',
+        email: '',
+        uid: '',
+      },
+    });
+  const [showResults, setShowResults] = useState<boolean>(false);
+
+  const [getRateLimits, rateLimitsResult] = useLazyQuery(GET_RATE_LIMITS);
+  const [unblockMutation] = useMutation<{ unblock: number }>(UNBLOCK);
```

Comment:
It'd be nice to make the variables typesafe too. e..g `useMutation<{unblock:number}, RateLimitQuery>(Unlbock)`. Same goes for other queries.